persistence-jdbc/scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/persistence/jdbc/JdbcPersistentEntityRegistry.scala

```diff
@@ -18,14 +17,14 @@ import com.lightbend.lagom.scaladsl.persistence.PersistentEntity
  * INTERNAL API
  */
 @Singleton
-private[lagom] final class JdbcPersistentEntityRegistry @Inject() (system: ActorSystem, injector: Injector, slickProvider: SlickProvider)
-  extends AbstractPersistentEntityRegistry(system, injector) {
+private[lagom] final class JdbcPersistentEntityRegistry @Inject() (system: ActorSystem, slickProvider: SlickProvider)
+  extends AbstractPersistentEntityRegistry(system) {
 
   private lazy val ensureTablesCreated = slickProvider.ensureTablesCreated()
 
-  override def register[C, E, S](entityClass: Class[_ <: PersistentEntity[C, E, S]]): Unit = {
+  override def register(entityFactory: => PersistentEntity[_, _, _]): Unit = {
     ensureTablesCreated
-    super.register(entityClass)
+    super.register(entityFactory)
```

Comment:
unrelated thought, could we provide a hook to override for first registration instead of each concrete class having to override register?